src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -520,4 +578,73 @@ void addUpper(Byte b, PreInitTrieNode child) {
             childrenUpper.put(b, child);
         }
     }
+
+    /**
```

Comment:
This JavaDoc is quite hard to understand without first understanding the full code. Maybe we can rewrite it so it clarifies a couple things? E.g.: what is "the sequence" and "repeated matching"?